proposals/0351-regex-builder.md

```diff
@@ -557,6 +558,56 @@ extension TryCapture {
 
 </details>
 
+### Mapping Output
+
+In addition to transforming individual captures within a regex, you can also map the output of an entire regex to a different output type. You can use the `mapOutput(_:)` methods to reorder captures, flatten nested optionals, or create instances of a custom type.
+
+This example shows how you can transform the output of a regex with three capture groups into an instance of a custom `SemanticVersion` type, matching strings such as `"1.0.0"` or `"1.0"`:
+
+```swift
+struct SemanticVersion: Equatable {
```

Comment:
```suggestion
struct SemanticVersion: Hashable {
```

The @lorentey conjecture: Anything that's substitutable under `Equatable` and is easily capable of being `Hashable` should be.